src/MultiFormats.Tests/MultiAddressTest.cs

```diff
@@ -0,0 +1,437 @@
+        }
+
+        [TestMethod]
+        public void Implicit_Conversion_From_String()
+        {
+            MultiAddress a = Somewhere;
+            Assert.IsInstanceOfType(a, typeof(MultiAddress));
+        }
+
+        [TestMethod]
+        public void Wire_Formats()
+        {
+            Assert.AreEqual(
+                new MultiAddress("/ip4/127.0.0.1/udp/1234").ToArray().ToHexString(),
+                "047f000001910204d2");
+            Assert.AreEqual(
+                new MultiAddress("/ip4/127.0.0.1/udp/1234/ip4/127.0.0.1/tcp/4321").ToArray().ToHexString(),
+                "047f000001910204d2047f0000010610e1");
+            Assert.AreEqual(
+                new MultiAddress("/ip6/2001:8a0:7ac5:4201:3ac9:86ff:fe31:7095").ToArray().ToHexString(),
+                "29200108a07ac542013ac986fffe317095");
+            Assert.AreEqual(
+                new MultiAddress("/ipfs/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSupNKC").ToArray().ToHexString(),
+                "a503221220d52ebb89d85b02a284948203a62ff28389c57c9f42beec4ec20db76a68911c0b");
+            Assert.AreEqual(
+                new MultiAddress("/ip4/127.0.0.1/udp/1234/utp").ToArray().ToHexString(),
+                "047f000001910204d2ae02");
+            Assert.AreEqual(
+                new MultiAddress("/onion/aaimaq4ygg2iegci:80").ToArray().ToHexString(),
+                "bc030010c0439831b48218480050");
+        }
+
+        [TestMethod]
+        public void PeerID_With_ipfs()
+        {
+            var ma = new MultiAddress("/ip4/10.1.10.10/tcp/29087/ipfs/QmVcSqVEsvm5RR9mBLjwpb2XjFVn5bPdPL69mL8PH45pPC");
+            Assert.AreEqual("QmVcSqVEsvm5RR9mBLjwpb2XjFVn5bPdPL69mL8PH45pPC", ma.PeerId.ToBase58());
+        }
+
+        [TestMethod]
+        public void PeerID_With_p2p()
+        {
+            var ma = new MultiAddress("/ip4/10.1.10.10/tcp/29087/p2p/QmVcSqVEsvm5RR9mBLjwpb2XjFVn5bPdPL69mL8PH45pPC");
+            Assert.AreEqual("QmVcSqVEsvm5RR9mBLjwpb2XjFVn5bPdPL69mL8PH45pPC", ma.PeerId.ToBase58());
+        }
+
+        [TestMethod]
+        public void PeerID_ipfs_p2p_are_equal()
+        {
+            var ipfs = new MultiAddress(
+                "/ip4/10.1.10.10/tcp/29087/ipfs/QmVcSqVEsvm5RR9mBLjwpb2XjFVn5bPdPL69mL8PH45pPC");
+            var p2P = new MultiAddress("/ip4/10.1.10.10/tcp/29087/p2p/QmVcSqVEsvm5RR9mBLjwpb2XjFVn5bPdPL69mL8PH45pPC");
+            Assert.AreEqual(ipfs, p2P);
+
+            var p2P1 = new MultiAddress("/ip4/10.1.10.10/tcp/29087/p2p/QmVCSqVEsvm5RR9mBLjwpb2XjFVn5bPdPL69mL8PH45pPC");
+            Assert.AreNotEqual(p2P, p2P1);
+
+            var p2P2 = new MultiAddress("/p2p/QmVcSqVEsvm5RR9mBLjwpb2XjFVn5bPdPL69mL8PH45pPC");
+            Assert.AreNotEqual(p2P, p2P2);
+        }
+
+        [TestMethod]
+        public void PeerID_Missing()
+        {
+            var ma = new MultiAddress("/ip4/10.1.10.10/tcp/29087");
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                var _ = ma.PeerId;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)